Segment_Tree_With_Lazi_Propagation/Segment_Tree_With_Lazy_Propagation.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * In this code we have a very large array called arr, and very large set of operations
+ * Operation #1: Increment the elements within range [i, j] with value val
+ * Operation #2: Get max element within range [i, j]
+ * Build tree: build_tree(1, 0, N-1)
+ * Update tree: update_tree(1, 0, N-1, i, j, value)
+ * Query tree: query_tree(1, 0, N-1, i, j)
```

Comment:
Your code looks good, but I would suggest you not to copy the code directly, from other sources (https://gist.github.com/shobhit6993/7088061) and try to implement it yourself.